<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>G3_31</class>
 <widget class="QWidget" name="G3_31">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>659</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>G3_31</string>
  </property>
  <widget class="QTextBrowser" name="textBrowser">
   <property name="geometry">
    <rect>
     <x>24</x>
     <y>10</y>
     <width>651</width>
     <height>111</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>20</pointsize>
    </font>
   </property>
   <property name="html">
    <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'幼圆'; font-size:20pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-family:'Microsoft YaHei UI'; font-size:9pt;&quot;&gt;题目：&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;建立一个名为&lt;/span&gt;&lt;span style=&quot; font-family:'Calibri','sans-serif'; font-size:10.5pt;&quot;&gt;Student&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;的类，该类有以下几个私有数据成员：学生姓名、学号、性别和年龄。还有以下两个成员函数：一个用于初始化学生姓名、学号、性别和年龄的构造函数，一个用于输出学生信息的函数。编写一个主函数，声明一个学生对象，然后调用成员函数在屏幕输出学生信息。修改类&lt;/span&gt;&lt;span style=&quot; font-family:'Calibri','sans-serif'; font-size:10.5pt;&quot;&gt;Student&lt;/span&gt;&lt;span style=&quot; font-family:'宋体'; font-size:10.5pt;&quot;&gt;，添加一个静态成员变量，用于表示已创建对象的数量：添加两个静态成员函数，一个用于输出已创建对象的数量，一个用于输出一个学生的姓名和学号。&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>226</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>学号：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>182</y>
     <width>51</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>姓名：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="id_1">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>222</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>1122334455</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>270</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>性别：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="name_1">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>178</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>张三</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="gender_1">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>266</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>男</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>314</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>年龄：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="age_1">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>310</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string>18</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="tian_jia">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>350</y>
     <width>151</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>确    定</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>140</y>
     <width>161</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>添  加  学  生</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>480</x>
     <y>620</y>
     <width>81</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>共计(人)：</string>
   </property>
  </widget>
  <widget class="QLabel" name="all_student">
   <property name="geometry">
    <rect>
     <x>560</x>
     <y>620</y>
     <width>81</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>0</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="Line" name="line">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>380</y>
     <width>201</width>
     <height>16</height>
    </rect>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QLabel" name="label_8">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>400</y>
     <width>161</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>查  找  学  生</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="sou_suo">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>630</y>
     <width>151</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>搜    索</string>
   </property>
  </widget>
  <widget class="QLabel" name="ti_shi">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>620</y>
     <width>161</width>
     <height>20</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
   </property>
  </widget>
  <widget class="QListView" name="student_new">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>180</y>
     <width>421</width>
     <height>431</height>
    </rect>
   </property>
  </widget>
  <widget class="QCheckBox" name="xx_name">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>430</y>
     <width>51</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>姓名</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="xx_id">
   <property name="geometry">
    <rect>
     <x>60</x>
     <y>430</y>
     <width>51</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>学号</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="xx_gender">
   <property name="geometry">
    <rect>
     <x>110</x>
     <y>430</y>
     <width>51</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>性别</string>
   </property>
  </widget>
  <widget class="QCheckBox" name="xx_age">
   <property name="geometry">
    <rect>
     <x>160</x>
     <y>430</y>
     <width>51</width>
     <height>20</height>
    </rect>
   </property>
   <property name="text">
    <string>年龄</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="age_2">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>590</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="id_2">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>502</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLineEdit" name="name_2">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>458</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>462</y>
     <width>51</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>姓名：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_11">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>506</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>学号：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_12">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>550</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>性别：</string>
   </property>
  </widget>
  <widget class="QLabel" name="label_13">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>594</y>
     <width>41</width>
     <height>16</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>年龄：</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="gender_2">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>546</y>
     <width>100</width>
     <height>25</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
  </widget>
  <widget class="QPushButton" name="all_all">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>140</y>
     <width>151</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>显 示 全 部</string>
   </property>
  </widget>
  <widget class="QPushButton" name="sou_sou">
   <property name="geometry">
    <rect>
     <x>500</x>
     <y>140</y>
     <width>151</width>
     <height>25</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>幼圆</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="text">
    <string>显 示 上 次 搜 索</string>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>tian_jia</sender>
   <signal>clicked()</signal>
   <receiver>G3_31</receiver>
   <slot>add_student()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>170</x>
     <y>367</y>
    </hint>
    <hint type="destinationlabel">
     <x>206</x>
     <y>360</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sou_suo</sender>
   <signal>clicked()</signal>
   <receiver>G3_31</receiver>
   <slot>search_student()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>166</x>
     <y>518</y>
    </hint>
    <hint type="destinationlabel">
     <x>215</x>
     <y>519</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xx_name</sender>
   <signal>clicked()</signal>
   <receiver>G3_31</receiver>
   <slot>xx_1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>18</x>
     <y>437</y>
    </hint>
    <hint type="destinationlabel">
     <x>13</x>
     <y>415</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xx_id</sender>
   <signal>clicked()</signal>
   <receiver>G3_31</receiver>
   <slot>xx_2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>80</x>
     <y>423</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xx_gender</sender>
   <signal>clicked()</signal>
   <receiver>G3_31</receiver>
   <slot>xx_3()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>133</x>
     <y>437</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>422</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xx_age</sender>
   <signal>clicked()</signal>
   <receiver>G3_31</receiver>
   <slot>xx_4()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>436</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>423</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>all_all</sender>
   <signal>clicked()</signal>
   <receiver>G3_31</receiver>
   <slot>all_all2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>368</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sou_sou</sender>
   <signal>clicked()</signal>
   <receiver>G3_31</receiver>
   <slot>sou_sou2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>621</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>671</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>add_student()</slot>
  <slot>search_student()</slot>
  <slot>xx_1()</slot>
  <slot>xx_2()</slot>
  <slot>xx_3()</slot>
  <slot>xx_4()</slot>
  <slot>all_all2()</slot>
  <slot>sou_sou2()</slot>
 </slots>
</ui>
